server/src/main/java/org/elasticsearch/index/stats/IndexingPressureStats.java

```diff
@@ -98,7 +105,8 @@ public IndexingPressureStats(
         long currentCoordinatingOps,
         long currentPrimaryOps,
         long currentReplicaOps,
-        long primaryDocumentRejections
+        long primaryDocumentRejections,
```

Comment:
Did we decide whether we wanted to be tracking operations (documents) or requests? I added these ops stats a very long time ago. We were talking about using them for calculating indexing time and back pressure. But then never went through that project. In fact we never even exposed these in node stats (https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html#cluster-nodes-stats-api-response-body-indexing-pressure).

This is why historically a "rejection" was request based opposed to ops based. I just want to make sure everyone is in agreement that we want to be calculating rejections based on individual ops opposed to requests.

Reply:
I think we want to track both.

I think visualizations should focus on the document level though.

Reply:
I changed PR. ES will emit the following metrics (counters) which concern about rejections. With help of this numbers it is possible to calculate rejection ratio on kibana using formula: ratio = rejected / (rejected + accepted) 

For coordination level (per request) rejections 
- es.indexing.coordinating_operations.rejections.total
- es.indexing.coordinating_operations.requests.total

For primary level (per document) rejections 
- es.indexing.primary_operations.total
- es.indexing.primary_operations.document.rejections.total

Reply:
Can these have their names being unified?

```
es.indexing.coordinating.requests.total
es.indexing.coordinating.requests.rejections.total

es.indexing.primary.documents.total
es.indexing.primary.documents.rejections.total
```

Reply:
I don't think we need "operations" if we are splitting it out by request vs. document. I think of operation as referring to an individual document.

Similarly we do not use the operations terminology in https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html.

Reply:
Can we defer naming change for a bit?

There are couple of prod dashboards which depends on current metric name e.g. on `es.indexing.primary_operations.total` and it is troublesome to change everything in one go.

I agree that naming should be more focused on "documents" and "requests" rather than just on "operations".